### PR TITLE
Feat/update helm charts

### PR DESCRIPTION
**Pull request by PaMarzec**
## Description
This PR updates the values.yaml and deployment.yaml in frontend and backend for their helm deployment.
Additionally  it adds a deployable umbrella helm chart for the [TRG 5.08](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-08/) requirement.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by tom-rm-meyer-ISST** [comment]
Dependency check will be solved in PR #42

File: charts/charts/puris-frontend/README.md.gotmpl [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 ## TL;DR
 ```shell
-$ helm install puris-frontend --namespace puris --create-namespace .
+$ helm install puris-frontend --generate-name
```
Comment (tom-rm-meyer-ISST) [inline charts/charts/puris-frontend/README.md.gotmpl:18]: update install command.

File: charts/charts/puris-frontend/templates/deployment.yaml [hunk 2]
```diff
@@ -34,10 +34,24 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           env:
-            - name: VITE_APP_NAME
-              value: "{{ .Values.app.viteAppName }}"
-            - name: VITE_BASE_URL
-              value: "{{ .Values.app.viteBaseUrl }}"
+            - name: APP_NAME
```
Comment (tom-rm-meyer-ISST) [inline charts/charts/puris-frontend/templates/deployment.yaml:37]: please change prefix from app to "puris-frontend". prefixes should follow the same convention. e.g. frontend and backend or even better puris-frontend and puris-backend

File: frontend/.env [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # Local Dev Environment
-VITE_APP_NAME=PURIS
```
Comment (tom-rm-meyer-ISST) [inline frontend/.env:2]: Please revert. These changes MUST NOT be applied to the .env.

The mechanism for docker is the following:
- .env has vite variables
- .env.dockerbuild has the vite variable that maps on an environment variable (VITE_BACKEND_BASE_URL=$BACKEND_BASE_URL)
- src/config.json has the environment variable names and the environment variable to substring in a json format.

When building the container:
- .env.dockerbuild is used
- vite / vue builds the application into a dest folder, that will be served by nginx
- result: for the env. VITE_BACKEND_BASE_URL it won't write a variable value BUT a placeholder into the built files ($BACKEND_BASE_URL)

When building the container, there is a "start-nginx.sh" file that does the following
- collect the environment variables (set for the docker container / set via helm as BACKEND_BASE_URL)
- Looks-up the "to replace string" from config.json (for BACKEND_BASE_URL, it will search for $BACKEND_BASE-URL in the built files)
- does the replacement in the built files
- starts nginx

File: frontend/.env.dockerbuild [hunk 4]
```diff
@@ -1,2 +1,2 @@
-VITE_APP_NAME=\$APP_NAME
-VITE_BACKEND_BASE_URL=\$BACKEND_BASE_URL
+APP_NAME=\$APP_NAME
```
Comment (tom-rm-meyer-ISST) [inline frontend/.env.dockerbuild:1]: revert, see above

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Overall great effort! There are smaller things and few bigger ones
- increase readability for operator: frontend and backend application environment variables should have a prefix
- installation command is not as good as previously
- .env changes are wrong - please revert

**Comment by tom-rm-meyer-ISST** [comment]
I added three new issues that shall cover improvements after this PR: 
- #46 
- #47 
- #48

File: charts/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
+# puris
```
Comment (tom-rm-meyer-ISST) [inline charts/README.md:1]: We need to adjust the overall chart structure. I had a mistake in mind:
charts
charts/puris
charts/puris/charts/backend
charts/puris/charts/frontend

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Please 
- fix KICS and Quality Checks. I left some points on how to do it. 
- update frontend environment variable usage
- adjust chart structure further as stated in charts/README.md conversation.

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, solves #35
